overkill_nvim/src/mapping/map.rs

```diff
@@ -0,0 +1,248 @@
+            mode,
+            arguments: SpecialArguments::default(),
+        }
+    }
+
+    /// # Panics
+    ///
+    /// This panics if `lhs` or `rhs` can't be converted to internal C-Strings (so don't pass any
+    /// `\0` chars!)
+    ///
+    pub fn map(&self, lhs: &str, rhs: &str) {
+        set_map(self.mode.as_str(), lhs, rhs, Some(self.arguments)).unwrap();
+    }
+
+    /// Yields `self` to `f`, but captures `self`'s arguments before, then sets them back after `f`
+    /// is called. It's helpful for reusing a `Mapper` for different purposes.
+    ///
+    /// ```no_run
+    /// use overkill_nvim::mapping::{MapMode, Mapper};
+    ///
+    /// let mut normal_mapper = Mapper::new(MapMode::Normal);
+    ///
+    /// // Use this group for normal+silent mappings.
+    /// normal_mapper.group(|mapper| {
+    ///     let mapper = mapper.silent();
+    ///     mapper.map("<F2>", "ihello<ESC>");
+    ///     mapper.map("<F4>", "cchello<ESC>");
+    /// });
+    ///
+    /// // Use this group for normal+unique mappings.
+    /// normal_mapper.group(|mapper| {
+    ///     let mapper = mapper.unique();
+    ///     mapper.noremap("<F3>", ":echo 'hi'<CR>");
+    /// });
+    ///
+    /// ```
+    ///
+    pub fn group<F>(&mut self, f: F)
+    where
+        F: Fn(Self),
+    {
+        let args_before = self.arguments;
+
+        f(*self);
+
+        self.arguments = args_before;
+    }
+
+    /// # Panics
+    ///
+    /// This panics if `lhs` or `rhs` can't be converted to internal C-Strings (so don't pass any
+    /// `\0` chars!)
+    ///
+    pub fn noremap(&self, lhs: &str, rhs: &str) {
+        set_noremap(self.mode.as_str(), lhs, rhs, Some(self.arguments)).unwrap();
+    }
+
+    /// # Panics
+    ///
+    /// This panics if `lhs` or `rhs` can't be converted to internal C-Strings (so don't pass any
+    /// `\0` chars!)
+    ///
+    pub fn buf_map(&self, buffer: Buffer, lhs: &str, rhs: &str) {
+        set_buf_map(buffer, self.mode.as_str(), lhs, rhs, Some(self.arguments)).unwrap();
+    }
+
+    /// # Panics
+    ///
+    /// This panics if `lhs` or `rhs` can't be converted to internal C-Strings (so don't pass any
+    /// `\0` chars!)
+    ///
+    pub fn buf_noremap(&self, buffer: Buffer, lhs: &str, rhs: &str) {
+        set_buf_noremap(buffer, self.mode.as_str(), lhs, rhs, Some(self.arguments)).unwrap();
+    }
+
+    def_bool_return_self_meth!(buffer);
+    def_bool_return_self_meth!(expr);
+    def_bool_return_self_meth!(nowait);
+    def_bool_return_self_meth!(script);
+    def_bool_return_self_meth!(silent);
+    def_bool_return_self_meth!(unique);
+
+    /// # Panics
+    ///
+    /// This will panic if the call to `nvim_api::keymaps::get_maps()` fails.
+    ///
+    #[must_use]
+    pub fn list_all(&self) -> Vec<Mapping> {
+        get_maps(self.mode.as_str())
+            .unwrap_or_else(|_| panic!("Unable to retrive maps for mode '{}'", self.mode.as_str()))
+            .into_iter()
+            .map(Mapping::from)
```

Comment:
I think this should filter out `Mapping`s that don't match the `Mapper`s options/args.